src/Microsoft.AspNetCore.Mvc.Core/Internal/AcceptHeaderParser.cs

```diff
@@ -80,12 +72,40 @@ private static bool TryParseValue(string value, ref int index, out MediaTypeSegm
                 return true;
             }
 
-            MediaTypeSegmentWithQuality result;
-            var length = GetMediaTypeWithQualityLength(value, currentIndex, out result);
+            // We deliverately want to ignore media types that we are not capable of parsing.
+            // This is due to the fact that some browsers will send invalid media types like 
+            // ; q=0.9 or */;q=0.2, etc.
+            // In this scenario, our recovery action consists of advancing the pointer to the
+            // next separator and moving on.
+            // In case we don't find the next separator, we simply advance the cursor to the
+            // end of the string to signal that we are done parsing.
+            MediaTypeSegmentWithQuality result = default(MediaTypeSegmentWithQuality);
+            int length = 0;
+            try
+            {
+                length = GetMediaTypeWithQualityLength(value, currentIndex, out result);
```

Comment:
Can you limit the try-catch to this method? None of the code after should throw, should it?

```
try
{
   length = ..
}
catch
{
  length = 0;
}
```